web/src/portico/integrations_dev_panel.ts

```diff
@@ -7,36 +9,83 @@ import * as channel from "../channel";
 // Data segment: We lazy load the requested fixtures from the backend
 // as and when required and then cache them here.
 
-const loaded_fixtures = new Map();
+const fixture_schema = z.record(
+    z.string(),
+    z.object({
+        body: z.unknown(),
+        headers: z.record(z.string()),
+    }),
+);
+
+type Fixutres = z.infer<typeof fixture_schema>;
+
+type HTMLSelectOneElement = HTMLSelectElement & {type: "select-one"};
+
+type ClearHandlers = {
+    stream_name: string;
+    topic_name: string;
+    URL: string;
+    results_notice: string;
+    bot_name: () => void;
+    integration_name: () => void;
+    fixture_name: () => void;
+    fixture_body: () => void;
+    custom_http_headers: () => void;
+    results: () => void;
+};
+
+const integrations_api_response_schema = z.object({
+    msg: z.string(),
+    responses: z.array(
+        z.object({
+            status_code: z.number(),
+            message: z.string(),
+            fixture_name: z.optional(z.string()),
+        }),
+    ),
+    result: z.string(),
+});
+
+type ServerResponse = z.infer<typeof integrations_api_response_schema>;
+
+const loaded_fixtures = new Map<string, Fixutres>();
 const url_base = "/api/v1/external/";
 
 // A map defining how to clear the various UI elements.
-const clear_handlers = {
+const clear_handlers: ClearHandlers = {
     stream_name: "#stream_name",
     topic_name: "#topic_name",
     URL: "#URL",
     results_notice: "#results_notice",
     bot_name() {
-        $("#bot_name").children()[0].selected = true;
+        const bot_options = $<HTMLSelectOneElement>(
+            "select:not([multiple])#bot_name",
+        ).children()[0];
+        assert(bot_options instanceof HTMLOptionElement);
+        bot_options.selected = true;
     },
     integration_name() {
-        $("#integration_name").children()[0].selected = true;
+        const integration_options = $<HTMLSelectOneElement>(
```

Comment:
nit: I think the plurality in the name isn't making much sense given we are accessing the first element with `children()[0]`. 
so - integration_option~s~ might be a better name??